diff --git a/src/wget.c b/src/wget.c
index b5e0f14..f6cb23e 100644
--- a/src/wget.c
+++ b/src/wget.c
@@ -1269,6 +1269,13 @@ static bool is_tty(void) {
 
 int main(int argc, const char **argv)
 {
+	#ifdef _WIN32
+		DWORD stdin_mode = 0;
+		DWORD stdout_mode = 0;
+		GetConsoleMode(GetStdHandle(STD_INPUT_HANDLE), &stdin_mode);
+		GetConsoleMode(GetStdHandle(STD_OUTPUT_HANDLE), &stdout_mode);
+		#endif
+
 	int n, rc;
 	char quota_buf[16];
 	long long start_time = 0;
@@ -1513,6 +1520,11 @@ int main(int argc, const char **argv)
 		program_deinit(); // destroy any resources belonging to this object file
 	}
 
+	#ifdef _WIN32
+		SetConsoleMode(GetStdHandle(STD_INPUT_HANDLE), stdin_mode);
+		SetConsoleMode(GetStdHandle(STD_OUTPUT_HANDLE), stdout_mode);
+	#endif
+
 	// Shutdown plugin system
 	plugin_db_finalize(get_exit_status());
 
